microceph/api/client_s3.go

```diff
@@ -0,0 +1,79 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/canonical/lxd/lxd/response"
+	"github.com/canonical/microceph/microceph/api/types"
+	"github.com/canonical/microceph/microceph/ceph"
+	"github.com/canonical/microcluster/rest"
+	"github.com/canonical/microcluster/state"
+)
+
+// /1.0/resources endpoint.
+var clientS3Cmd = rest.Endpoint{
+	Path:   "client/s3",
+	Get:    rest.EndpointAction{Handler: cmdClientS3Get, ProxyTarget: true},
+	Put:    rest.EndpointAction{Handler: cmdClientS3Put, ProxyTarget: true},
+	Delete: rest.EndpointAction{Handler: cmdClientS3Delete, ProxyTarget: true},
+}
+
+func cmdClientS3Get(s *state.State, r *http.Request) response.Response {
```

Comment:
RESTfulness: 
- retrieving a list of users should be a GET to `/client/s3`
- retrieving a single user should be a GET to `/client/s3/{userid}`

See the /disks/{osdid} handler in microceph/api/disks.go for an example on how to implement a handler for a path variable i.e. {userid}